tests/unit/test_string.cpp

```diff
@@ -87,15 +87,25 @@ void check_single_string(File file, size_t string_length) {
         auto obj =
             CreateTraits::create(file, "overlength_nullterm", dataspace, overlength_nullterm);
         obj.write(value);
-        REQUIRE(obj.template read<std::string>() == value);
+        if (rest_vol_enabled()) {
+            const auto readValue = obj.template read<std::string>();
+            REQUIRE(value.compare(0, readValue.size(), readValue) == 0);
+        } else {
+            REQUIRE(obj.template read<std::string>() == value);
+        }
```

Comment:
We could maybe wrap this in a lambda since it show up many times.

```suggestion
        auto check = [&](const auto &obj, const auto &value){
        if (rest_vol_enabled()) {
            const auto readValue = obj.template read<std::string>();
            REQUIRE(value.compare(0, readValue.size(), readValue) == 0);
        } else {
            REQUIRE(obj.template read<std::string>() == value);
        }
        }
        check(obj, value);
```

Reply:
The problem is that this is inter-mingled between catch2 SECTIONS, so the lambda is not ideal to share. I will leave it as-is for now.